src/text/encoder.rs

```diff
@@ -0,0 +1,264 @@
+    /// Writes a string value to `output` as an Ion symbol.
+    pub fn encode_str_as_symbol<S: AsRef<str>, W: io::Write>(
+        output: &mut W,
+        value: S,
+    ) -> IonResult<()> {
+        write!(output, "'{}'", value.as_ref())?;
+        Ok(())
+    }
+
+    /// Writes a string value to `output` as an Ion string.
+    pub fn encode_str_as_string<S: AsRef<str>, W: io::Write>(
+        output: &mut W,
+        value: S,
+    ) -> IonResult<()> {
+        write!(output, "\"{}\"", value.as_ref())?;
+        Ok(())
+    }
+
+    /// Writes a byte array slice to `output` as an Ion clob.
+    pub fn encode_clob<W: io::Write>(_output: &mut W, _value: &[u8]) -> IonResult<()> {
+        todo!()
+    }
+
+    /// Writes a byte array slice to `output` as an Ion blob.
+    pub fn encode_blob<W: io::Write>(output: &mut W, value: &[u8]) -> IonResult<()> {
+        // Rust format strings escape curly braces by doubling them. The following string is:
+        // * The opening {{ from a text Ion blob, with each brace doubled to escape it.
+        // * A {} pair used by the format string to indicate where the base64-encoded bytes
+        //   should be inserted.
+        // * The closing }} from a text Ion blob, with each brace doubled to escape it.
+        // TODO: Provide a re-usable encoding buffer instead of allocating a String each time.
+        write!(output, "{{{{{}}}}}", base64::encode(value))?;
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::result::IonResult;
+    use crate::text::encoder::TextEncoder;
+    use crate::IonType;
+    use bigdecimal::BigDecimal;
+    use chrono::{FixedOffset, NaiveDate, TimeZone};
+    use std::str;
+    use std::str::FromStr;
+
+    fn encoder_test<F>(mut commands: F, expected: &str)
+    where
+        F: FnMut(&mut Vec<u8>) -> IonResult<()>,
+    {
+        let mut output = Vec::new();
+        commands(&mut output).expect("Invalid TextEncoder commands.");
+        assert_eq!(str::from_utf8(&output).unwrap(), expected);
+    }
+
+    #[test]
+    fn encode_null() {
+        encoder_test(
+            |output| TextEncoder::encode_null(output, IonType::Struct),
+            "null.struct",
+        );
+    }
+
+    #[test]
+    fn encode_bool() {
+        encoder_test(|output| TextEncoder::encode_bool(output, true), "true");
+        encoder_test(|output| TextEncoder::encode_bool(output, false), "false");
+    }
+
+    #[test]
+    fn encode_i64() {
+        encoder_test(|output| TextEncoder::encode_i64(output, -99999), "-99999");
+        encoder_test(|output| TextEncoder::encode_i64(output, -42), "-42");
+        encoder_test(|output| TextEncoder::encode_i64(output, -1), "-1");
+        encoder_test(|output| TextEncoder::encode_i64(output, 0), "0");
+        encoder_test(|output| TextEncoder::encode_i64(output, 1), "1");
+        encoder_test(|output| TextEncoder::encode_i64(output, 42), "42");
+        encoder_test(|output| TextEncoder::encode_i64(output, 99999), "99999");
+    }
```

Comment:
Not a blocking concern, but you might want to take a look at some of my use of [`rstest`][1] here.

[1]: https://github.com/amzn/ion-rust/blob/c1479ea3f09432f34254d385eba91fb95af236ec/ion-c-sys/src/lib.rs#L466-L614